Test/GTestPluginManager.cpp

```diff
@@ -169,34 +170,48 @@ TEST(TestPluginManagerPathResolution, RespectsAbsolutePath_Config)
 
 TEST(TestPluginManagerPathResolution, RespectsEnvironmentOverride_Plugin)
 {
-    setenv("PLUGIN_PATH", "/tmp/overridden", 1);
+    std::filesystem::path const overriddenPath("/tmp/overridden");
+    std::filesystem::create_directories(overriddenPath);
+
+    auto const exePath = plugin::detail::GetApplicationPath();
+
+    std::filesystem::copy_file(exePath / "plugins" / "libDummyGeneratorPlugin.so", overriddenPath / "some.plugin.so", std::filesystem::copy_options::overwrite_existing);
+
+    setenv("PLUGIN_PATH", overriddenPath.c_str(), 1);
 
     EXPECT_THAT(plugin::PluginManager::ResolvePluginPath("some.plugin.so"),
                 Eq("/tmp/overridden/some.plugin.so"));
 }
 
-TEST(TestPluginManagerPathResolution, RespectsEnvironmentOverride_Config)
+TEST(TestPluginManagerPathResolution, RespectsEnvironmentOverride_Plugin_Multipath)
 {
-    setenv("PLUGIN_CONFIG_PATH", "/tmp/overridden_config", 1);
+    std::filesystem::path const overriddenPath("/tmp/overridden");
+    std::filesystem::create_directories(overriddenPath);
 
-    EXPECT_THAT(plugin::PluginManager::ResolveConfigPath("some.pluginlist.pb.txt"),
-                Eq("/tmp/overridden_config/some.pluginlist.pb.txt"));
-}
+    auto const exePath = plugin::detail::GetApplicationPath();
 
-TEST(TestPluginManagerPathResolution, RespectsEnvironmentOverrideWithRelativePath_Plugin)
-{
-    setenv("PLUGIN_PATH", "/tmp/overridden", 1);
+    std::filesystem::copy_file(exePath / "plugins" / "libDummyGeneratorPlugin.so", overriddenPath / "some.plugin.so", std::filesystem::copy_options::overwrite_existing);
 
-    EXPECT_THAT(plugin::PluginManager::ResolvePluginPath("../sub1/some.plugin.so"),
-                Eq("/tmp/overridden/../sub1/some.plugin.so"));
+    auto const multiPath = "/tmp/does_not_exist1:/tmp/does_not_exist2:"s + overriddenPath.native() + ":/tmp/does_not_exist3";
+    setenv("PLUGIN_PATH", multiPath.c_str(), 1);
+
+    EXPECT_THAT(plugin::PluginManager::ResolvePluginPath("some.plugin.so"),
+                Eq("/tmp/overridden/some.plugin.so"));
 }
 
-TEST(TestPluginManagerPathResolution, RespectsEnvironmentOverrideWithRelativePath_Config)
```

Comment:
I removed these tests, they no longer work, which is a good thing in my opinion -- I don't think that allowing relative paths to some other arbitrary location on the filesystem within a plugin configuration file was a good idea to begin with.